packages/storage/src/types.ts

```diff
@@ -1,12 +1,13 @@
 export type StorageWithOptions<O> = {
-  clear: () => void;
+  clear: (options?: O) => void;
   getItem: (key: string, options?: O) => string | null;
-  getAll?: () => { [key: string]: any };
+  getAll?: (options?: O) => { [key: string]: any };
   setItem: (key: string, value: string, options?: O) => void;
-  removeItem: (key: string) => void;
-  key: (index: number) => string | null;
+  removeItem: (key: string, options?: O) => void;
+  key: (index: number, options?: O) => string | null;
+  getLength: (options?: O) => number | undefined;
   readonly length: number | undefined;
-  [key: string]: any;
+  [name: string]: any;
```

Comment:
Since in the Storage API, the standard calls them keys, I prefer the previous version.

Reply:
yeah, therefore everything is optional, i forgot to make getLength as whole optional, now fixed. But as mentioned before we need that, so that we can if we want pass the options to everywhere it is needed

Reply:
Thanks for working with me on improving this.

Reply:
ups, overlooked, now fixed